statistics/queries/continent_histogram.sql

```diff
@@ -18,61 +19,60 @@ dl_per_location AS (
-dl_stats_perip_perday AS (
+
+# Gather rows per day and ip
+dl_rows_perip_perday AS (
   SELECT
     date,
     continent_code,
     ip,
-    MIN(mbps) AS download_MIN,
-    APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(25)] AS download_Q25,
-    APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(50)] AS download_MED,
-    AVG(mbps) AS download_AVG,
-    APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(75)] AS download_Q75,
-    MAX(mbps) AS download_MAX,
-    APPROX_QUANTILES(MinRTT, 100) [SAFE_ORDINAL(50)] AS download_minRTT_MED
+    # This organizes all tests, in an arbitrary but repeatable order.
+    ARRAY_AGG(STRUCT(ABS(FARM_FINGERPRINT(id)) AS ffid, mbps, MinRTT) ORDER BY ABS(FARM_FINGERPRINT(id))) AS members,
+    STRUCT(
+      ROUND(MIN(mbps),3) AS download_MIN,
+      ROUND(APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(25)],3) AS download_Q25,
+      ROUND(APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(50)],3) AS download_MED,
+      ROUND(POW(10,AVG(Safe.LOG10(mbps))),3) AS download_LOG_AVG,
+      ROUND(AVG(mbps),3) AS download_AVG,
+      ROUND(APPROX_QUANTILES(mbps, 100) [SAFE_ORDINAL(75)],3) AS download_Q75,
+      ROUND(MAX(mbps),3) AS download_MAX,
+      ROUND(APPROX_QUANTILES(MinRTT, 100) [SAFE_ORDINAL(50)],3) AS download_minRTT_MED
+    ) AS stats
   FROM dl_per_location_cleaned
   GROUP BY date, continent_code, ip
+  --ORDER BY ip, date
 ),
-# Calculate final stats per day from 1x test per ip per day normalization in prev. step
-dl_stats_per_day AS (
+
+dl_random_perip_perday AS (
   SELECT
     date,
     continent_code,
-    MIN(download_MIN) AS download_MIN,
-    APPROX_QUANTILES(download_Q25, 100) [SAFE_ORDINAL(25)] AS download_Q25,
-    APPROX_QUANTILES(download_MED, 100) [SAFE_ORDINAL(50)] AS download_MED,
-    AVG(download_AVG) AS download_AVG,
-    APPROX_QUANTILES(download_Q75, 100) [SAFE_ORDINAL(75)] AS download_Q75,
-    MAX(download_MAX) AS download_MAX,
-    APPROX_QUANTILES(download_minRTT_MED, 100) [SAFE_ORDINAL(50)] AS download_minRTT_MED
-  FROM
-    dl_stats_perip_perday
-  GROUP BY date, continent_code
+    ip,
+    ARRAY_LENGTH(members) AS tests,
+    # Use a prime number larger than the number of total rows in an aggregation to select a random row 
+    members[SAFE_OFFSET(MOD(511232941,ARRAY_LENGTH(members)))] AS random1,
+    members[SAFE_OFFSET(MOD(906686609,ARRAY_LENGTH(members)))] AS random2,
+    stats
+  FROM dl_rows_perip_perday
 ),
-# Count the difference in the number of tests from the same IPs on the same
-#   day, to the number of tests used in the daily statistics.
-dl_samples_total AS (
+
+# Calculate log average per day from random samples
+dl_stats_per_day AS (
   SELECT
-    COUNT(*) AS dl_total_samples,
-    date,
-    continent_code
-  FROM dl_per_location_cleaned
-  GROUP BY
-    date,
-    continent_code
+    date, continent_code,
+    ROUND(POW(10,AVG(Safe.LOG10(stats.download_MED))),3) AS dl_day_log_avg_median,
+    ROUND(POW(10,AVG(Safe.LOG10(random1.MinRtt))),3) AS dl_min_rtt_day_log_avg,
```

Comment:
Calculate log avg twice for these metrics, using random1 and random2

Reply:
Done in [d516ad7](https://github.com/m-lab/stats-pipeline/pull/42/commits/d516ad7bdad668b0b71f348e532838ab21b1c438)